core-kafka-streams/src/main/java/org/galatea/kafka/starter/messaging/KafkaStreamsConfig.java

```diff
@@ -4,28 +4,32 @@
 import java.util.Map;
 import java.util.Properties;
 import lombok.Getter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
 
-public class StreamProperties {
+@Component
+@ConfigurationProperties
```

Comment:
Suggest following the convention of using a prefix, rather than making other modules worry about namespace conflicts.
Consider:
prefix = kafka-streams
Map<String, String> stream-properties
Or something slightly less verbose.

Reply:
so in that case, the properties in the yml would look like:
`
kafka-streams.stream-properties:
  application.id: app
  bootstrap.servers: host:9092
`
I'm not a big fan of the redundant layer "stream-properties"...

Reply:
Spring's documented convention around using a prefix is motivated by the idea that adding additional properties to your config class shouldn't gobble more of the namespace.
I suppose if you pinky-swear that you'll never need any properties besides StreamProperties for your core module (dubious), it's... fine, but you still have the aesthetic weirdness of having your config module effectively scanning the whole config map.
I'll approve the PR, but give it a think.